script.js

```diff
@@ -0,0 +1,104 @@
+//1
+/*const calcMinNumber = (number1, number2, number3, number4) => {
+    const calcResult = Math.min(number1, number2, number3, number4);
+    if (isNaN(calcResult) || calcResult === null || calcResult === undefined) {
+      return "You entered invalid variables";
+    } else {
+      return calcResult;
+    }
+  };
+  calcMinNumber(number1, number2, number3, number4);
+  */
+//2
+function greatestDivisor(x, y) {
+    if (typeof x !== "number" || typeof y !== "number") {
+      console.log("Please enter two numbers");
+      return "Please enter two numbers";
+    } else if (!Number.isInteger(x) || !Number.isInteger(y)) {
+      console.log("Please enter whole numbers");
+      return "Please enter whole numbers";
+    } else {
+      x = Math.abs(x);
+      y = Math.abs(y);
+      while (y) {
+        let c = y;
+        y = x % y;
+        x = c;
+      }
+      console.log(x);
+      return x;
+    }
+  }
+  
+  greatestDivisor(x, y);
+  
+  //3
+  function perfectNumber(number) {
+    let startNumber = 0;
+    for (let i = 1; i <= number / 2; i++) {
+      if (number % i === 0) {
+        startNumber += i;
+      }
+    }
+  
+    if (startNumber === number && startNumber !== 0) {
+      return "It is a perfect number.";
+    } else {
+      return "It is not a perfect number.";
+    }
+  }
+  
+  perfectNumber(number);
+  
+  //4
+  let firstValue = 10;
+  let lastValue = -2;
```

Comment:
```suggestion
  const lastValue = -2;
```

Reply:
And we need those values predefined? We would call function with given arguments in there, but not the case they are already predefined inside code